state/upgrades.go

```diff
@@ -2266,10 +2266,15 @@ func AddControllerNodeDocs(pool *StatePool) error {
 			Insert: doc,
 		})
 	}
-	if len(ops) > 0 {
-		return errors.Trace(st.runRawTransaction(ops))
-	}
-	return nil
+
+	ops = append(ops, txn.Op{
```

Comment:
This will always run the update, even if there's nothing to do - is it idempotent? I guess it probably is since if the machineids field is renamed then it does nothing.

Reply:
yup, is idempotent, and the test runs the step twice